tests/phpunit/tests/l10n.php

```diff
@@ -431,6 +431,15 @@ public function test_length_of_excerpt_rss_should_be_counted_by_chars() {
 	 * @covers ::wp_dashboard_recent_drafts
 	 */
 	public function test_length_of_draft_should_be_counted_by_words() {
+		if ( PHP_VERSION_ID >= 80100 ) {
+			/*
+			 * On the one hand, this test would throw a PHP 8.1 "null to non-nullable" deprecation,
+			 * on the other hand, when that deprecation is set to be expected, the test still fails.
+			 * This needs further investigation.
+			 */
+			$this->markTestSkipped( 'Needs investigation.' );
+		}
+
```

Comment:
```suggestion
		/*
		 * Set the current user to Editor so that the get_edit_post_link() call
		 * in wp_dashboard_recent_drafts() works as expected and returns a string
		 * instead of null.
		 *
		 * This is a prerequisite for wp_dashboard_recent_drafts(), which is only
		 * called in core for users with the 'edit_posts' capability.
		 *
		 * This also avoids a PHP 8.1 deprecation notice when the result of
		 * get_edit_post_link() is passed to esc_url() further in the function:
		 * `ltrim(): Passing null to parameter #1 ($string) of type string is deprecated`.
		 */
		wp_set_current_user( self::factory()->user->create( array( 'role' => 'editor' ) ) );

```
@jrfnl After some investigation, this is what I had in mind for these tests instead of skipping.

`wp_dashboard_recent_drafts()` is only called in core for users with the `edit_posts` capability, so adding that missing prerequisite makes the test pass without errors and without any changes to core. It still tests the draft length calculation, as originally intended. Same for the other two tests.

Reply:
@SergeyBiryukov Nice one! Sounds to me like a good solution.

I have verified that the current user is reset after the test(s) via the `WP_UnitTestCase_Base::tear_down()` method, so this can be committed as is.

I will replace the test skip commit with a commit for the above.